cluster-autoscaler/cloudprovider/clusterapi/clusterapi_unstructured.go

```diff
@@ -304,6 +304,17 @@ func (r unstructuredScalableResource) InstanceCapacity() (map[corev1.ResourceNam
 	return capacity, nil
 }
 
+// InstanceSystemInfo sets the nodeSystemInfo from the infrastructure reference resource.
+// If the infrastructure reference resource is not found, returns nil.
+func (r unstructuredScalableResource) InstanceSystemInfo() *apiv1.NodeSystemInfo {
+	infraObj, err := r.readInfrastructureReferenceResource()
```

Comment:
something i am concerned about is the notion that we will be calling this function twice for every node template we create. each call to this function results in another call to the api server to get the infrastructure machine template. i think we need to find a way to combine the call required in this function with the call we make in the `InstanceCapacity` call.

Reply:
To keep having two separate functions and avoid coupling, I've made a change to the readInfrastructureReferenceResource to cache the infraObj via lazy loading. Wdyt?